File: backend/rust_server/src/models/building.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;

#[derive(Debug, Serialize, Deserialize, FromRow, Clone, ToSchema)]
pub struct Building {
    pub building_id: i32,                  // 건물 ID (Primary Key)
    pub node_id: String,                  // 노드 ID
    pub name: String,                     // 건물 이름
    pub alias: Option<String>,            // 별칭 (Optional)
    pub category: Option<String>,         // 카테고리 (Optional)
    pub basement_floors: i32,             // 지하 층수
    pub ground_floors: i32,               // 지상 층수
    pub roof_floors: i32,                 // 옥상 층수
    pub address: String,                  // 주소
    pub phone_number: Option<String>,     // 전화번호 (Optional)
    pub tags: Option<String>,             // 태그 (Optional, 예: "편의점, 휠체어 충전소, 카페")
}

File: frontend/moducnu/lib/domain/model/building.dart
class Building {
  final String? address; // 건물 주소
  final String? alias; // 건물 별칭
  final int? basementFloors; // 지하 층 수
  final int buildingId; // 건물 ID
  final String? category; // 건물 카테고리
  final int? groundFloors; // 지상 층 수
  final String name; // 건물 이름
  final String nodeId; // 노드 ID
  final String? phoneNumber; // 전화번호
  final int? roofFloors; // 옥상 층 수
  final List<String>? tags; // 태그 리스트
  final String disabledRestrooms; // 장애인 화장실 정보 (문자열)
  final String elevators; // 엘리베이터 정보 (문자열)
  final String ramps; // 경사로 정보 (문자열)
  final List<FloorPlan>? floorPlans; // 독립적인 층별 도면 데이터

  Building({
    this.address,
    this.alias,
    this.basementFloors,
    required this.buildingId,
    this.category,
    this.groundFloors,
    required this.name,
    required this.nodeId,
    this.phoneNumber,
    this.roofFloors,
    this.tags,
    required this.disabledRestrooms,
    required this.elevators,
    required this.ramps,
    this.floorPlans,
  });
}

class FloorPlan {
  final int floorNumber; // 층 번호
  final String? imagePath; // 도면 이미지 경로

  FloorPlan({
    required this.floorNumber,
    this.imagePath,
  });
}
